### PR TITLE
Update SVG component to support based64

### PR DESCRIPTION
**Pull request by ChaoLiangSuper**
Resolves: #756 

- Update svg component to support base 64 encoded
- Add polyfill for svg component
- [x] Fix test
- [x] Update docs

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Check-list:

- [x] I have read the [Contributing](CONTRIBUTING.md) document.
- [ ] I've thought about and labelled my PR/commit message appropriately.
- [ ] If this PR introduces breaking changes I've described the impact and migration path for existing applications.
- [x] CI is green (coverage, linting, tests).
- [x] I have updated the documentation accordingly.
- [ ] I've two LGTMs/Approvals.
- [ ] I've fixed or replied to all my code-review comments.
- [ ] I've manually tested with a buddy.
- [x] I've squashed my commits into one.

### CODE REVIEW

File: src/components/alexandria/SvgSymbol/index.jsx [hunk 1]
```diff
@@ -3,8 +3,15 @@ import PropTypes from 'prop-types';
 import { classSuffixHelper } from 'lib/utils';
 import './styles.scss';
 
+const componentName = 'AlexandriaSvgSymbolComponent';
+
 const SvgSymbol = props => {
-  const { classSuffixes, href, onClick } = props;
+  if (!props.href && !props.based64Data) {
+    console.error(`One of props 'href' or 'based64Data' must be specified in '${componentName}'.`);
```
Comment (nimishjha) [inline src/components/alexandria/SvgSymbol/index.jsx]: I don't think we need to check for this and show an error. If they provide nothing, it renders nothing and it should be obvious what the problem is -- and we have example code for every component.

File: src/components/alexandria/SvgSymbol/index.jsx [hunk 2]
```diff
@@ -13,18 +20,29 @@ const SvgSymbol = props => {
...
   );
 };
 
-SvgSymbol.displayName = 'AlexandriaSvgSymbolComponent';
+SvgSymbol.displayName = componentName;
 
 SvgSymbol.propTypes = {
   classSuffixes: PropTypes.arrayOf(PropTypes.string.isRequired),
   href: PropTypes.string,
+  based64Data: PropTypes.string,
```
Comment (nimishjha) [inline src/components/alexandria/SvgSymbol/index.jsx]: Didn't we discuss using custom function `propTypes` to check for the situation where at least one of two props is required?

File: src/components/alexandria/SvgSymbol/index.jsx [hunk 3]
```diff
@@ -3,7 +3,10 @@ import PropTypes from 'prop-types';
 import { classSuffixHelper } from 'lib/utils';
 import './styles.scss';
 
+const base64Reg = /data:image\/svg\+xml;base64,.+/g;
```
Comment (nimishjha) [inline src/components/alexandria/SvgSymbol/index.jsx]: This regex is too lenient. Base64 data only contains @, =, and 0-9A-Za-z. I'm searching for a better regex, will let you know.

Comment (nimishjha) [reply src/components/alexandria/SvgSymbol/index.jsx]: Use this instead: 

`^@[a-zA-Z0-9+/]+={0,2}$`

Comment (nimishjha) [reply src/components/alexandria/SvgSymbol/index.jsx]: in place of the `.+`

Comment (ChaoLiangSuper) [reply src/components/alexandria/SvgSymbol/index.jsx]: As discussed offline, will use `indexOf`

**Pull request merged**